test(elacticity with nemesis): test 90 percent with supported nemesis

yarongilor opened the pull request:
test supported nemesis with 90 percent
Task: https://github.com/scylladb/scylla-cluster-tests/issues/9155
Full nemesis status details:
please see full details in https://github.com/scylladb/scylla-cluster-tests/issues/9155#issuecomment-2535262107
### Testing
<!-- Add links to Argus/Jenkins of test test done with this PR -->
<!-- This would help the reviewer to cross check what was tested, and and review the results as needed -->
- [new pipeline test](https://argus.scylladb.com/tests/scylla-cluster-tests/2504ce8f-17b9-4f17-9914-bbf3313bec0f)

### PR pre-checks (self review)
<!--- PR should be created as Draft, when CI finished and relevant checkboxes selected, add reviewers and then click on "Ready for review" button.-->
<!--- Put an `x` in all the boxes that apply or create PR and then click on all relevant checkboxes: -->
- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

yarongilor commented:
TODO: check if this PR also requires updating nemesis unit-test following the new added label.

yarongilor commented:
The [test](https://argus.scylladb.com/tests/scylla-cluster-tests/46216cb9-31c1-4a79-8644-4fa9f7b12f33) of 40 hours passed ok overall.
It had one nemesis failure with a new scylla [issue](https://github.com/scylladb/scylladb/issues/22464).
It got 2 very small reactor stalls (that indicated a known issue).

yarongilor commented on test-cases/features/elasticity-90-percent-with-nemesis.yaml:
```diff
@@ -0,0 +1,35 @@
+nemesis_grow_shrink_instance_type: 'i4i.large'
+
+instance_type_loader: 'c6i.2xlarge'
+instance_type_monitor: 't3.large'
```
Not sure about setting this `instance_type_monitor`, but no harm with it, and probably keeps consistent with similar other tests.

yarongilor commented:
@scylladb/qa-maintainers please merge if no comments.

soyacz commented on sdcm/nemesis.py:
```diff
@@ -213,6 +213,7 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
     # i.e. adding/removing nodes/data centers
     disruptive: bool = False        # flag that signal that nemesis disrupts node/cluster,
     # i.e reboot,kill, hardreboot, terminate
+    elasticity: bool = False  # supported in a 90% disk utilization scenario
```
I'm not sure if `elasticity` is a good name for it.
`high_disk_utilization` or something that would reflect the use case

yarongilor replied:
well, the official feature name is "elastic-cloud", that's the reason. we can change to it if it sounds clearer.

soyacz replied:
elastic-cloud is still enigmatic for newcomers. Anyway it's just my opinion.
And after second thought I'd rather give it `supports_high_disk_utilization`

yarongilor replied:
ok, fixed by the following changes:
```
-    elasticity: bool = False  # supported in a 90% disk utilization scenario
+    supports_high_disk_utilization: bool = True  # supported in a 90% disk utilization scenario
```
```
+    supports_high_disk_utilization = False  # Decommissioning a node consumes disk space
+    supports_high_disk_utilization = False  # Snapshot/Restore operations consume disk space
+    supports_high_disk_utilization = False  # Snapshot/Restore operations consume disk space
+    supports_high_disk_utilization = False  # Snapshot/Restore operations consume disk space
+    supports_high_disk_utilization = False  # Removing a node consumes disk space
+    supports_high_disk_utilization = False  # Failed for: https://github.com/scylladb/scylladb/issues/22088
+    supports_high_disk_utilization = False  # Failed for: https://github.com/scylladb/scylladb/issues/22088
+    supports_high_disk_utilization = False  # Creating an Index consumes disk space
+    supports_high_disk_utilization = False  # Creating an MV consumes disk space
```
```
-nemesis_selector: ['elasticity']
+nemesis_selector: ['supports_high_disk_utilization']
```

soyacz commented on data_dir/nemesis.yml:
```diff
@@ -206,10 +228,7 @@
   - run_with_gemini = False
   - kubernetes = True
   - limited = True
-- disrupt_nodetool_refresh:
```
@yarongilor why this nemesis got removed? (same for enospc)

yarongilor replied:
@soyacz , because it was erroneously duplicated.

soyacz reviewed (commented):
I don't see why some nemesis are not supported - aren't nemesis world simulation we try to test and make sure all works/properly recovers to normal state?

If that's not the case would be worth to add `elasticyt = False` for these which are not supported showing the reason.

yarongilor commented:
> I don't see why some nemesis are not supported - aren't nemesis world simulation we try to test and make sure all works/properly recovers to normal state?
> 
> If that's not the case would be worth to add `elasticyt = False` for these which are not supported showing the reason.

@soyacz , please see full details in https://github.com/scylladb/scylla-cluster-tests/issues/9155#issuecomment-2535262107

soyacz commented:
> > I don't see why some nemesis are not supported - aren't nemesis world simulation we try to test and make sure all works/properly recovers to normal state?
> > If that's not the case would be worth to add `elasticyt = False` for these which are not supported showing the reason.
> 
> @soyacz , please see full details in [#9155 (comment)](https://github.com/scylladb/scylla-cluster-tests/issues/9155#issuecomment-2535262107)

this is exactly what I'd expect - for each `False` value, attach inline issue (so it can be enabled once fixed) or just 'causes enospc error (which is ok for things like create mv)`

soyacz reviewed (approved):
LGTM

yarongilor commented:
@pehala , configuration paths changed to:
```
	renamed:    jenkins-pipelines/oss/features/elasticity-90-percent-with-nemesis.jenkinsfile -> jenkins-pipelines/oss/features/elasticity/elasticity-90-percent-with-nemesis.jenkinsfile
	renamed:    test-cases/features/elastic-cloud/elasticity-90-percent-with-nemesis.yaml -> test-cases/features/elasticity/elasticity-90-percent-with-nemesis.yaml

```

soyacz commented:
@yarongilor @pehala do we need it in 2025.1?

pehala commented:
> @yarongilor @pehala do we need it in 2025.1?

No
